k8s/pkg/registryserver/nse.go

```diff
@@ -53,11 +54,23 @@ func (rs *nseRegistryService) RegisterNSE(ctx context.Context, request *registry
 			logrus.Errorf("Failed to register nsm: %s", err)
 			return nil, err
 		}
-		nseResponse, err := rs.cache.AddNetworkServiceEndpoint(&v1.NetworkServiceEndpoint{
-			ObjectMeta: metav1.ObjectMeta{
-				GenerateName: request.GetNetworkService().GetName(),
+
+		var objectMeta metav1.ObjectMeta
+		if request.GetNetworkserviceEndpoint().EndpointName == "" {
+			NSEPrefix := strings.Join([]string{rs.nsmName, request.GetNetworkService().GetName()}, "-")
```

Comment:
Since you have handled case with similar nse's names, is it still necessary?

Reply:
Agreed, lets return more readable name